workflow/controller/workflowpod.go

```diff
@@ -455,21 +455,13 @@ func (woc *wfOperationCtx) podExists(nodeID string) (existing *apiv1.Pod, exists
 		return nil, false, fmt.Errorf("failed to get pod from informer store: %w", err)
 	}
 
-	objectCount := len(objs)
-
-	if objectCount == 0 {
+	if len(objs) == 0 {
 		return nil, false, nil
 	}
 
-	if objectCount > 1 {
-		return nil, false, fmt.Errorf("expected < 2 pods, got %d - this is a bug", len(objs))
```

Comment:
Do we know the root cause for this though?

Reply:
This happens when there is another workflow recently run with same name.